### ISSUE TITLE
The KNN approximation method

### ISSUE THREAD

**Issue by Thomas-Galtier** [SubstraFoundation/distributed-learning-contributivity#39]
Hello,
I've find an interesting method that  seems to run at least 10³ times faster than the  truncated monte-carlo method! Check this paper [An Empirical and Comparative Analysis of Data Valuation with Scalable Algorithms](https://arxiv.org/pdf/1911.07128.pdf)  Ruoxi Jia, Xuehui Sun, Jiacen Xu, Ce Zhang, Bo Li, Dawn Song . 2019

In this paper they approximate a neural network by fitting a k-nearest-neighbour algorithm to the deep features (the layer before the last softmax layer). Then they easily and fastly compute the shapley value for this approximation of the neural network, enjoying that the shapley value for a k-nearest-neighbour model is analytically computable. The empirical results of the paper show that This method goes at least 10³ times faster than the  truncated monte-carlo method! Though this method is heuristic and yet has no theoretical justification, the paper shows that, in practice, the method identifies well the corrupted and beneficial data.
Note that the results of this paper and of a previous paper suggest their approximated shapley values are roughly ranked like the true shapley values, but the paper does not tell if the proposed method yields a good approximation of the true shapley value. We should check if it is the case.

**Comment by Thomas-Galtier** [comment]
I contacted one of the autors of the article and it seems that the method does not work on convolutive neural network. Here is a graph of an exemple that shows the estimations are pretty bad.
![KNN vs true shapley](https://user-images.githubusercontent.com/57995084/72205376-73b73200-3482-11ea-836c-e90797de3f2b.png)
